src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -72,14 +74,30 @@ public DefaultCaseDataAccessControl(RoleAssignmentService roleAssignmentService,
     // for the user and filters access profiles based on the case type.
     @Override
     public Set<AccessProfile> generateAccessProfilesByCaseTypeId(String caseTypeId) {
+        return generateAllTypesOfProfilesByCaseTypeId(caseTypeId, false);
+    }
+
+    @Override
+    public Set<AccessProfile> generateCreationAccessProfilesByCaseTypeId(String caseTypeId) {
+        return generateAllTypesOfProfilesByCaseTypeId(caseTypeId, true);
+    }
+
+    private Set<AccessProfile> generateAllTypesOfProfilesByCaseTypeId(String caseTypeId, boolean isACreationProfile) {
+
         CaseTypeDefinition caseTypeDefinition = caseDefinitionRepository.getCaseType(caseTypeId);
-        RoleAssignments roleAssignments = roleAssignmentService.getRoleAssignments(securityUtils.getUserId());
+        final RoleAssignments roleAssignments;
+        if (isACreationProfile) {
+            roleAssignments = roleAssignmentService.getRoleAssignmentsForCreate(securityUtils.getUserId());
```

Comment:
instead of new method getRoleAssignmentsForCreate, can we overload getRoleAssignments(string) to getRoleAssignments(string, boolean)

Reply:
I think that is better if can separate the logic of filtering organisation and take advance of the method name.  For that reason I added a new method.